tests/fail/concurrency/windows_join_main.rs

```diff
@@ -0,0 +1,28 @@
+//@only-target-windows: Uses win32 api functions
+// We are making scheduler assumptions here.
+//@compile-flags: -Zmiri-preemption-rate=0
+
+// On windows, joining main is not UB, but it will block a thread forever.
+
+use std::thread;
+
+extern "system" {
+    fn WaitForSingleObject(handle: isize, timeout: u32) -> u32;
+}
+
+const INFINITE: u32 = u32::MAX;
+
+// This is how miri represents the handle for thread 0.
```

Comment:
```suggestion
// XXX HACK: This is how miri represents the handle for thread 0.
```

Nice test :)

Reply:
Relying on implementation details is always fun :)